src/handlers/bulk/prepare/handler.js

```diff
@@ -106,17 +107,70 @@ const bulkPrepare = async (error, messages) => {
 
     const { hasDuplicateId, hasDuplicateHash } = await Comparators.duplicateCheckComparator(bulkTransferId, payload, BulkTransferService.getBulkTransferDuplicateCheck, BulkTransferService.saveBulkTransferDuplicateCheck)
     if (hasDuplicateId && hasDuplicateHash) {
-      Logger.isErrorEnabled && Logger.error(Util.breadcrumb(location, `callbackErrorModified--${actionLetter}1`))
+      const eventDetail = { functionality: Enum.Events.Event.Type.NOTIFICATION, action: TransferEventAction.PREPARE_DUPLICATE }
+
+      // Check to see if the resend is coming from a participant of involved
+      // with the bulk transfer to avoid potential data leakage.
+      const participants = await BulkTransferService.getParticipantsById(bulkTransferId)
+      if (![participants.payeeFsp, participants.payerFsp].includes(message.value.from)) {
```

Comment:
This one and the next check should be unreachable code, isn't it? "hasDuplicateHash" should indicate that the headers and payload are exactly same as before.. So the check that this is coming from the same Payer FSP as before should already be covered there..